aesd-char-driver/aesd-circular-buffer.c

```diff
@@ -62,7 +62,7 @@ if(buffer!=NULL){
     }
     buffer->entry[buffer->in_offs].buffptr=add_entry->buffptr;
     buffer->entry[buffer->in_offs].size=add_entry->size;
-
+    buffer->len+=add_entry->size;
```

Comment:
@Sona-Subair if you are going to track the buffer len you'll need to subtract when removing items from the buffer as well.  When the buffer is full you'll need to remove the size of the entry you replaced.